test/string-parser/parse-auction.test.ts

```diff
@@ -0,0 +1,56 @@
+import { Types } from '../../src';
+import { parseAuction } from '../../src/string-parser';
+import { PossibleCalls } from '../../src/types';
+
+const stringContestedAuction =
+	'1S-2D-3D-P-5C-P-5D-P-5NT-P-6H-P-7S-X-P-P-XX-P-P-P';
+const stringUncontestedAuction = '1H-2C-2H-2N-3C-3NT';
+
+const listContestedAuction = [
+	{ call: { level: 1, suit: Types.Suit.Spade } },
+	{ call: { level: 2, suit: Types.Suit.Diamond } },
+	{ call: { level: 3, suit: Types.Suit.Diamond } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 5, suit: Types.Suit.Club } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 5, suit: Types.Suit.Diamond } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 5, suit: Types.NoTrump } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 6, suit: Types.Suit.Heart } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 7, suit: Types.Suit.Spade } },
+	{ call: PossibleCalls.Double },
+	{ call: PossibleCalls.Pass },
+	{ call: PossibleCalls.Pass },
+	{ call: PossibleCalls.Redouble },
+	{ call: PossibleCalls.Pass },
+	{ call: PossibleCalls.Pass },
+	{ call: PossibleCalls.Pass },
+];
+const listUncontestedAuction = [
+	{ call: { level: 1, suit: Types.Suit.Heart } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 2, suit: Types.Suit.Club } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 2, suit: Types.Suit.Heart } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 2, suit: Types.NoTrump } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 3, suit: Types.Suit.Club } },
+	{ call: PossibleCalls.Pass },
+	{ call: { level: 3, suit: Types.NoTrump } },
```

Comment:
Hmm, this poses the question of an uncontested auction ending with three passes. e.g. `1H-2H-P` you'd probably expect to have three passes at the end. Maybe the solution is to always add a final pass to the end of uncontested auctions. Come to think of it you may want some way of starting with 1 initial pass in an uncontested auction, which isn't currently possible.

Reply:
Yeah this was one of the edge cases I was worried about. I dont think we need to force an auction to have 3 passes at the end here, because we may want partial auctions e.g. when posing problems?

Reply:
TODO add comment that it returns an extra pass at the end